number_test.go

```diff
@@ -584,3 +590,177 @@ func TestNumber_ConvertLesser(t *testing.T) {
 	value.chain.assertFailed(t)
 	value.chain.clearFailed()
 }
+
+func TestNumber_IsInt(t *testing.T) {
+	reporter := newMockReporter(t)
+
+	t.Run("check parameters", func(t *testing.T) {
+		NewNumber(reporter, 1234).IsInt(0, 1, 2, 3).
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234).IsInt().
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, 1234.0001).NotInt(0, 1, 2, 3).
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234.0001).NotInt().
+			chain.assertNotFailed(t)
+	})
+
+	t.Run("without bit size", func(t *testing.T) {
+		NewNumber(reporter, -1234).IsInt().
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, 1234.00001).IsInt().
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234).NotInt().
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234.00001).NotInt().
+			chain.assertNotFailed(t)
+	})
+
+	t.Run("with bit size", func(t *testing.T) {
+		NewNumber(reporter, -math.MaxInt8).IsInt(32).
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, math.MaxInt64).IsInt(32).
+			chain.assertFailed(t)
+
+		NewNumber(reporter, math.MaxInt64).NotInt(32).
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, math.MaxInt8).NotInt(32).
+			chain.assertFailed(t)
+	})
+
+	t.Run("check edge cases", func(t *testing.T) {
```

Comment:
Let's ensure that +inf and -inf is covered for both IsInt and NotInt in this subtest.

Also, I'd rename this test to something like "non-numbers".

Reply:
Similar for IsUnit test.